src/compiler.ts

```diff
@@ -6214,6 +6214,37 @@ export class Compiler extends DiagnosticEmitter {
         // fall-through
       }
 
+      case ElementKind.INDEXSIGNATURE: {
```

Comment:
The fall-through above, from the CLASS case into the INDEXSIGNATURE case, seems suspicious.

Reply:
Nope. Not at all.

Reply:
Thanks. There is also the resolver's `resolveCallExpression` that needs to be updated with an `INDEXSIGNATURE` case accordingly, so it mirrors the compiler's `compileCallExpression`.

Reply:
Sorry for the delay; I wasn't able to access my laptop for about a month. I notice that `compileCallExpression` has more cases than `resolveCallExpression`, so is such a change necessary?

Reply:
I've recently looked at the PR again, and found that there is likely potential for general improvements. All the cases that compile the target expression can probably be unified to in fact compile the target expression (call a local, call a global, call a property, call an index signature), then evaluate the type of it, get the signature, and do the indirect call. However, #2548 changes a bunch of code in close vicinity, so I haven't managed yet to make changes to this PR, which I originally intended.

Suggesting to wait for after the mentioned PR has landed, and then see? Would you like to look into these simplifications yourself?